packages/toolpad-app/src/server/data.ts

```diff
@@ -455,3 +455,13 @@ export async function loadDom(appId: string, version: VersionOrPreview = 'previe
 export async function loadRenderTree(appId: string, version: VersionOrPreview = 'preview') {
   return appDom.createRenderTree(await loadDom(appId, version));
 }
+
+export async function duplicateApp(id: string): Promise<AppMeta> {
+  const app = await prismaClient.app.findUnique({ where: { id } });
```

Comment:
Isn't this check already happening in `loadPreviewDom`?

Reply:
I feel like that adds an unnecessary step and is a UX downgrade. The standard expectation with a duplicate/"make a copy" interaction is not a dialog box from what I can see across most benchmarks. 

I propose we could add an assertion to satisfy the type system: `app!.name ...`

Reply:
I think being taken to the duplicated app directly from the list page might be an overreach:

On the list page, on duplication, I would expect that the overall view remains consistent and another app appears in the list with an indication to it being the duplicated app.  In this case it is the "(copy)" added to the name. We could potentially even add a selected outline/focus indicator to it.

Bechmarking with the Google Drive experience:

https://user-images.githubusercontent.com/19550456/189125012-62122cc6-7584-4ada-921b-ca2251783417.mov

"Dialog box -> Redirected to the new app" is an okay expectation for me when I am inside the "editor" itself; in the future, if we allow duplication while someone is inside the editor, I think we should follow the dialog box UX (We could potentially include other options along with the title in that case.)

Again, benchmarking with Google Drive:

https://user-images.githubusercontent.com/19550456/189125657-78e91f6c-f10e-4621-b0ca-98c3fc21abd4.mov

I don't think the rename action is that immediately important; and the pain of having to go back to the list view to rename should be mitigated by adding this inline renaming functionality in the future:

https://user-images.githubusercontent.com/19550456/189129657-10ec5bbe-7679-4b6b-8eb8-acac00052ad7.mov

Reply:
In any case, both implementations can be built by passing the name to `duplicateApp`